src/pages/auth/Login.jsx

```diff
@@ -1,9 +1,79 @@
-import React from "react";
+import React, { useRef, useState, useEffect } from "react";
 import "./auth.css";
 
-import { Link } from "react-router-dom";
+import { Link, Navigate, useLocation, useNavigate } from "react-router-dom";
+import { useAxios } from "../../utils/useAxios";
+import { useAuth } from "../../context";
+
+import { validateEmail, validatePassword } from "./utils/inputValidateUtils";
 
 const Login = () => {
+  const [showPassword, setShowPassword] = useState(false);
+  const initialFormValidateState = {
+    isEmailValid: true,
+    isPasswordValid: true,
+  };
+  const [formValidateState, setFormValidateState] = useState(
+    initialFormValidateState
+  );
+
+  const emailRef = useRef();
+  const passwordRef = useRef();
+
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const { isLoggedIn, setEncodedToken } = useAuth();
+  const { makeRequest, response, error } = useAxios();
+
+  const loginHandler = () => {
+    setFormValidateState((prev) => initialFormValidateState);
+    if (
+      validateEmail(emailRef.current.value) &&
+      validatePassword(passwordRef.current.value)
+    ) {
+      makeRequest({
+        method: "post",
+        url: "/api/auth/login",
+        data: {
+          email: emailRef.current.value,
+          password: passwordRef.current.value,
+        },
+      });
+    } else {
+      if (!validateEmail(emailRef.current.value)) {
+        setFormValidateState((prev) => ({ ...prev, isEmailValid: false }));
+      }
+      if (!validatePassword(passwordRef.current.value)) {
+        setFormValidateState((prev) => ({ ...prev, isPasswordValid: false }));
+      }
+    }
+  };
+
+  const guestLoginHandler = () => {
+    emailRef.current.value = "adarshbalika@gmail.com";
+    passwordRef.current.value = "adarshBalika123";
+    loginHandler();
+  };
+
+  useEffect(() => {
+    if (response) {
+      setEncodedToken(response.encodedToken);
+      localStorage.setItem("encodedToken", response.encodedToken);
+    }
+    if (error) {
+      passwordRef.current.value = "";
```

Comment:
Suggestion: You can directly use a state (instead of useRef) here as you are changing the value too.